pkg/apis/pipeline/v1alpha1/contexts.go

```diff
@@ -28,3 +28,25 @@ type hdcnKey struct{}
 func WithDefaultConfigurationName(ctx context.Context) context.Context {
 	return context.WithValue(ctx, hdcnKey{}, struct{}{})
 }
+
+// HasDefaultConfigurationName checks to see whether the given context has
+// been marked as having a default configurationName.
+func HasDefaultConfigurationName(ctx context.Context) bool {
+	return ctx.Value(hdcnKey{}) != nil
+}
+
+// lemonadeKey is used as the key for associating information
+// with a context.Context.
+type lemonadeKey struct{}
+
+// WithUpgradeViaDefaulting notes on the context that we want defaulting to rewrite
+// from v1alpha1 to v1beta1.
+func WithUpgradeViaDefaulting(ctx context.Context) context.Context {
+	return context.WithValue(ctx, lemonadeKey{}, struct{}{})
+}
+
+// IsUpgradeViaDefaulting checks whether we should be "defaulting" from v1alpha1 to
+// the v1beta1 subset.
```

Comment:
I'm a bit confused about what `v1alpha1` and `v1beta1` are referring to in this context - my understanding is that we are going from pre 0.5.0 to 0.5.0 ? Or does `v1beta1` here refer to the version of a lib we are using? (maybe the comment could clarify?)

Reply:
ah lol, I write that "ahead" of time. it should be from pr 0.5.0 to post 0.5.0 indeed :wink: 
I need to update the comment indeed :+1: